src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs

```diff
@@ -30,17 +30,21 @@ public class WebFunctionsManager : IWebFunctionsManager
         private const string DurableTaskStorageConnectionName = "azureStorageConnectionStringName";
         private const string DurableTask = "durableTask";
 
-        private readonly ScriptJobHostOptions _config;
+        private readonly ScriptJobHostOptions _hostOptions;
         private readonly ILogger _logger;
         private readonly HttpClient _client;
         private readonly IEnumerable<WorkerConfig> _workerConfigs;
+        private readonly IOptions<HttpOptions> _httpOptions;
+        private readonly IProxyMetadataManager _proxyMetadataManager;
 
-        public WebFunctionsManager(IOptions<ScriptApplicationHostOptions> webSettings, IOptions<LanguageWorkerOptions> workerConfigOptions, ILoggerFactory loggerFactory, HttpClient client)
+        public WebFunctionsManager(IOptions<ScriptApplicationHostOptions> applicationHostOptions, IOptions<LanguageWorkerOptions> languageWorkerOptions, IOptions<HttpOptions> httpOptions, ILoggerFactory loggerFactory, HttpClient client, IProxyMetadataManager proxyMetadataManager)
```

Comment:
`IWebFunctionsManager` is a long lived service registered at the root, taking an `HttpOptions` would be problematic as the options setup is handled by the extension, which lives in the job host scope.

At the very least, we need to take in an `IOptionsMonitor` and make sure we're using the `CurrentValue` (or subscribing for change notifications on the monitor). We still have a challenge with the fact that our host.json configuration source is JobHost scoped and the setup can't just rely on configuration if it doesn't move. If it does, then we have to wire things up for change notification as the configuration will need to be updated if the file changes. :(

is this only being used to get the route prefix?

Reply:
Chose to read the http prefix value from host.json anew each time for this API. This API is already reading function.json files so this change is in line with that. I marked that with a TODO so if we come up with a better lifetime scoping solution we can change.